### PR TITLE
Add a markmap mind map page for scaling InnerSource

### PR DESCRIPTION
**Pull request by jeffabailey**
An output of an @InnerSourceCommons/ispo-working-group meeting: https://docs.google.com/document/d/12PFkoiaukmrMbnTPYKERzViKCULfClutXN4ydVPv3H0/edit

feat: add markmap partial
feat: add markmap shortcode
feat: add markmap friendly markmap content
fix: skip yourmax processing if no items defined

### CODE REVIEW

**Comment by dellagustin-sap** [comment]
Hi @jeffabailey , I tested the PR locally and found two issues:
1. Major - the /learn page is showing some "weird" stuff, screenshot below, it shows the markdown text of the mindmap under "Scaling InnerSource" and it shows also an additional section with the content of `markmap_content.md`, but slightly different.
2. Minor - On the scaling page, the mindmap has a kind of a viewport, where it can be zoomed in and out, panned and scrolled, but without scrollbars or boarders - this is more a weird feature than a bug, just wanted to mention

I found a way to fix 1 and I'll send a PR on top of this PR with a fix.

image

image

**Comment by dellagustin-sap** [comment]
@jeffabailey , here is the proposed fix: https://github.com/InnerSourceCommons/innersourcecommons.org/pull/902

File: content/en/learn/scaling/mindmap.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+title: "Scaling InnerSource"
+subtitle: "How can I scale InnerSource?"
+type: "community"
+image: "images/learn/patterns.png"
+weight: 1
+---
+
+{{< markmap path="markmap_content.md" >}}
```
Comment (dellagustin-sap) [inline content/en/learn/scaling/mindmap.md]: @jeffabailey all the other files on the repo use a `-` as separator.
I suggest renaming `markmap_content.md` to `markmap-content.md` for consistency.

**Comment by dellagustin-sap** [comment]
Also, it would be nice if we could make the links open in a new tab, but I think that would require some implementation.
Here are some refs:
- https://github.com/markmap/markmap/blob/master/packages/markmap-autoloader/README.md
- https://github.com/markmap/markmap/issues/129

by adding a transformer plugin with something like this, we may be able to do it:
```
<script>
  window.markmap = {
    autoLoader: {
      transformPlugins: [...markmap.builtInPlugins, { 
     name: 'target-blank', 
     transform(transformHooks) { 
       transformHooks.parser.tap((md) => { 
         md.renderer.renderAttrs = wrapFunction( 
           md.renderer.renderAttrs, 
           (renderAttrs, token) => { 
             let attrs = renderAttrs(token); 
             if (token.type === 'link_open') { 
               attrs += ' target="_blank"'; 
             } 
             return attrs; 
           }, 
         ); 
       }); 
       return {}; 
     }, 
   }],
    },
  };
</script>
```

I'll need to test this later.

**Comment by dellagustin-sap** [comment]
LGTM, but I'm not a maintainer.
A minor quirk, viewing on my 15" macbook display, the viewport of the mindmap does not fit the screen in 100% zoom, vertically.
image

**Comment by jeffabailey** [comment]
Unless we compact the mindmap, I'm afraid that's an avoidable quirk.

However, assuming that all operating systems and browsers allow you to hold ctrl to pan the mindmap in any direction, we could put a quick instruction above the mindmap.

**Comment by dellagustin-sap** [comment]
> However, assuming that all operating systems and browsers allow you to hold ctrl to pan the mindmap in any direction, we could put a quick instruction above the mindmap.

I was thinking about that too, some instructions on top would be nice. I don't think we need to hold the merge for that though, that could be an increment with another PR, unless you want to include it already, up to you @jeffabailey .

One minor ask, can you please link to https://docs.google.com/document/d/12PFkoiaukmrMbnTPYKERzViKCULfClutXN4ydVPv3H0/edit on the description of the PR? Just for additional traceability on the origin of the mindmap.

**Comment by jeffabailey** [comment]
Hello @InnerSourceCommons/website-maintainers do you have any concerns with merging this PR?

If no, please approve. :)

**Review by CiaraFlanagan — APPROVED** [review]

**Pull request merged**